### PR TITLE
Migrations as another project

### PR DESCRIPTION
**Pull request by lastland**
In this approach, migrations and application are separated in two sbt sub projects. We don't really need an sbt plugin to run the migration commands -- we just need to run the other sub project, and we can easily define an alias for that. We don't really need to change much code in this approach.

This approach reduces sbt dependency significantly (comparing with #2 ). Changes in migration sub projects do not require an sbt reload, either.

### CODE REVIEW

File: example/build.sbt [hunk 1]
```diff
@@ -10,14 +10,4 @@ scalacOptions += "-deprecation"
...
-  ,"com.h2database" % "h2" % "1.3.166"
-  ,"org.xerial" % "sqlite-jdbc" % "3.6.20"
-  ,"org.slf4j" % "slf4j-nop" % "1.6.4" // <- disables logging
-/*
-  // enables logging
-  ,"org.slf4j" % "slf4j-api" % "1.6.4"
-  ,"ch.qos.logback" % "logback-classic" % "0.9.28"
-*/
-)
+addCommandAlias("mg", "migrations/run")
```
Comment (cvogt) [inline example/build.sbt:13]: whats the idea with customized migration tools? Do people create a project that depends on our migration tool as a dependency and setup an alias to direct to that project's run? Seems sensible.

File: example/migrations/src/main/scala/SampleCodegen.scala [hunk 2]
```diff
@@ -17,20 +17,22 @@ object SampleCodegen{
...
           tableNames.contains(t.name.name))))
       val latest = mm.latest
-      List( "v" + latest, "latest" ).foreach{ version =>
-        val pkg = "datamodel." + version + ".schema"
-        val folder = System.getProperty("user.dir")+"/app/src/main/scala"
-        val generator = new SourceCodeGenerator(model) {
-          override def packageCode(
-            profile: String, pkg: String, container: String) : String =
-            super.packageCode(profile, pkg, container) + s"""
+      List("/app", "/migrations").foreach { proj =>
```
Comment (cvogt) [inline example/migrations/src/main/scala/SampleCodegen.scala:20]: do you generate the stuff into two folders? some project dependencies should allow to have one generation target, right?

Comment (lastland) [inline example/migrations/src/main/scala/SampleCodegen.scala:20]: Yes I generate the stuff into two folders. It's just a temporary workaround.

What do you mean? Generating a distinct project and have the application project and migrations project depend on it?

File: example/build.sbt (hunk 1 above)
Comment (lastland) [inline example/build.sbt:13]: Exactly.

File: example/migrations/src/main/scala/SampleCodegen.scala (hunk 2 above)
Comment (cvogt) [inline example/migrations/src/main/scala/SampleCodegen.scala:20]: temporary sounds good.
Here is the project relationships I imagine:
project migrations depends on project generated code
project example app depends on project generated code
project migration manager runs project migrations, which triggers code generation into project generated code (only). it doesn't depend on project migrations or project generated code, which means if either of them are broken you can still run the migration manager to e.g. delete broken generated code

File: example/build.sbt (hunk 1 above)
Comment (cvogt) [inline example/build.sbt:13]: great :)!

**Pull request merged**